From: Tom Coughlan <coughlan@redhat.com>
Subject: [RHEL 5.0 PATCH] QLogic qla2xxx - add missing PCI device IDs
Date: Wed, 13 Dec 2006 17:59:34 -0500
Bugzilla: 219350
Message-Id: <1166050775.26511.98.camel@bianchi.boston.redhat.com>
Changelog: QLogic qla2xxx - add missing PCI device IDs


>From QLogic:

qla2xxx does not properly setup the scsi_host_template for ISP54xx
type HBAs (this includes qla220 and qle220 boards).  The attached
patch corrects this oversight.

BZ 219350.

I built it okay. QLogic (and likely IBM and EMC) tested it. Not upstream
yet, but will be soon.

Tom

diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index 65cbe2f..1e0e984 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -1350,7 +1350,9 @@ qla2x00_probe_one(struct pci_dev *pdev, const struct pci_device_id *id)
 
 	sht = &qla2x00_driver_template;
 	if (pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2422 ||
-	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2432)
+	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2432 ||
+	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP5422 ||
+	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP5432)
 		sht = &qla24xx_driver_template;
 	host = scsi_host_alloc(sht, sizeof(scsi_qla_host_t));
 	if (host == NULL) {

